modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/DangerModal.tsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayModal from '@clayui/modal';
+import {Observer} from '@clayui/modal/lib/types';
+import React, {useState} from 'react';
+
+import Input from './Form/Input';
+
+export default function DangerModal({
+	children,
+	errorMessage,
+	observer,
+	onClose,
+	onDelete,
+	title,
+	token,
+}: IProps) {
+	const [value, setValue] = useState<string>();
+
+	return (
+		<ClayModal observer={observer} status="danger">
+			<ClayModal.Header>{title}</ClayModal.Header>
+
+			<ClayModal.Body>
+				{children}
+
+				<Input
+					error={
+						value !== undefined &&
+						value.toLowerCase() !== token.toLowerCase()
+							? errorMessage
+							: ''
+					}
+					onChange={({target: {value}}: any) => {
+						setValue(value);
+					}}
```

Comment:
Avoid using the `any` type, be as strict as possible and move all type definitions to the bottom of the file.

Here, I believe you can omit the type and the TypeScript will infer the proper type, since the `onChange` has already it defined:
```suggestion
					onChange={({target: {value}}) => setValue(value)}
```